src/sentry/models/dashboard.py

```diff
@@ -34,12 +35,22 @@ class DashboardFavoriteUser(DefaultFieldsModel):
     __relocation_scope__ = RelocationScope.Organization
 
     user_id = HybridCloudForeignKey("sentry.User", on_delete="CASCADE")
+    organization = FlexibleForeignKey("sentry.Organization", null=True)
     dashboard = FlexibleForeignKey("sentry.Dashboard", on_delete=models.CASCADE)
 
+    position = models.PositiveSmallIntegerField(null=True, db_default=None)
+
     class Meta:
         app_label = "sentry"
         db_table = "sentry_dashboardfavoriteuser"
-        unique_together = (("user_id", "dashboard"),)
+        constraints = [
+            # A user can only favorite a dashboard once
+            UniqueConstraint(
+                fields=["user_id", "dashboard"],
+                name="sentry_dashboardfavoriteuser_unique_favorite_per_dashboard",
+                deferrable=models.Deferrable.DEFERRED,
+            ),
+        ]
```

Comment:
Are we changing this to deferred so that it works well inside of the transaction?